msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerContentProviderStrategy.java

```diff
@@ -0,0 +1,487 @@
+                return mRequestAdapter.getRequestStringForGetAccounts(parameters);
+            }
+
+            @Nullable
+            @Override
+            public String[] getOptions() {
+                return new String[]{negotiatedBrokerProtocolVersion};
+            }
+
+            @NonNull
+            @Override
+            public String getMethodName() {
+                return methodName;
+            }
+
+            @NonNull
+            @Override
+            public String getUriPath() {
+                return GET_ACCOUNTS_PATH;
+            }
+
+            @NonNull
+            @Override
+            public List<ICacheRecord> getResultFromBundle(Bundle resultBundle) throws BaseException {
+                return mResultAdapter.getAccountsFromResultBundle(resultBundle);
+            }
+        });
+    }
+
+    @Override
+    void removeBrokerAccount(@NonNull final RemoveAccountCommandParameters parameters,
+                             @Nullable final String negotiatedBrokerProtocolVersion) throws BaseException {
+
+        final String methodName = "removeBrokerAccountWithContentProvider";
+
+        performContentProviderOperation(
+                parameters,
+                new ContentProviderOperation<RemoveAccountCommandParameters, Void>() {
+            @Nullable
+            @Override
+            public String getRequestString(RemoveAccountCommandParameters parameters) {
+                return mRequestAdapter.getRequestStringForRemoveAccount(parameters);
+            }
+
+            @Nullable
+            @Override
+            public String[] getOptions() {
+                return new String[]{negotiatedBrokerProtocolVersion};
+            }
+
+            @NonNull
+            @Override
+            public String getMethodName() {
+                return methodName;
+            }
+
+            @NonNull
+            @Override
+            public String getUriPath() {
+                return REMOVE_ACCOUNTS_PATH;
+            }
+
+            @NonNull
+            @Override
+            public Void getResultFromBundle(Bundle resultBundle) throws BaseException {
+                mResultAdapter.verifyRemoveAccountResultFromBundle(resultBundle);
+                return null;
+            }
+        });
+    }
+
+    @Override
+    boolean getDeviceMode(@NonNull final CommandParameters parameters,
+                          @Nullable final String negotiatedBrokerProtocolVersion) throws BaseException {
+        final String methodName = "getDeviceModeWithContentProvider";
+        return performContentProviderOperation(
+                parameters,
+                new ContentProviderOperation<CommandParameters, Boolean>() {
+
+            @Nullable
+            @Override
+            public String getRequestString(CommandParameters parameters) {
+                return null; // broker returns a boolean to indicate the device mode, no request string needed.
```

Comment:
Having `null` seems confusing and seems like nothing is being sent to the broker. I believe this is NULL because the only thing we are sending is basically telling the broker that we need to get the device mode? and due to the nature of the get device mode command, it doesn't need any additional parameters to be sent along unlike other operations such as remove account?

Reply:
Yes, `requestString` is different from `requestOperation` in itself. There could be operations to broker which doesn't need any additional request data to be passed , like `getDeviceMode` here. But the interface is common for all the operations , hence I tagged this with `Nullable`